crates/rspack_plugin_javascript/src/visitors/dependency/parser/walk_block_pre.rs

```diff
@@ -30,7 +30,7 @@ impl<'parser> JavascriptParser<'parser> {
           }
           ModuleDecl::ExportNamed(decl) => self.block_pre_walk_export_name_declaration(decl),
           ModuleDecl::ExportDefaultExpr(_) => (),
```

Comment:
seems `export default function Mod() {}` will have the same problem?

Reply:
> seems `export default function Mod() {}` will have the same problem?

`ExportDefaultDeclaration` case is covered here:

https://github.com/web-infra-dev/rspack/blob/ffd46f01b5985f8ae16dd36ab4faffc4a262741c/crates/rspack_plugin_javascript/src/visitors/dependency/parser/walk_block_pre.rs?plain=1#L28-L30

https://github.com/web-infra-dev/rspack/blob/ffd46f01b5985f8ae16dd36ab4faffc4a262741c/crates/rspack_plugin_javascript/src/visitors/dependency/parser/walk_block_pre.rs?plain=1#L93-L123

Reply:
test case added: https://github.com/web-infra-dev/rspack/pull/6024